contracts/src/v0.8/keystone/KeystoneFeedsPermissionHandler.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {OwnerIsCreator} from "../shared/access/OwnerIsCreator.sol";
+
+/// @title Keystone Feeds Permission Handler
+/// @notice This contract is designed to manage and validate permissions for accessing specific reports within a decentralized system.
+/// @dev The contract uses mappings to keep track of report permissions associated with a unique report ID.
+abstract contract KeystoneFeedsPermissionHandler is OwnerIsCreator {
```

Comment:
This is only used for the `KeystoneFeedsConsumer`, we generally do not extract functionality into a contract for single use. It makes it harder to reason about and harder to read the final contract, as logic is split over multiple files.

Reply:
> This is only used for the KeystoneFeedsConsumer

If I recall correctly `KeystoneFeedsConsumer` is an example contract, and may or may not be used in production 

The base contract here however is `KeystoneFeedsPermissionHandler` which abstracts out the functionality of maintaining reporting permissions for Keystone Feeds taking into account the forwarder, workflowName, reportName, workflowOwner. 

This in combination with keystone's `IReceiver` interface can help contracts implement their own version of `onReport`